pkg/image/prune/imagepruner.go

```diff
@@ -53,12 +67,59 @@ type ImagePruner interface {
 
 // imagePruner implements ImagePruner.
 type imagePruner struct {
-	g         graph.Graph
-	algorithm pruneAlgorithm
+	g              graph.Graph
+	algorithm      pruneAlgorithm
+	registryPinger registryPinger
 }
 
 var _ ImagePruner = &imagePruner{}
 
+// registryPinger performs a health check against a registry.
+type registryPinger interface {
+	// ping performs a health check against registry.
+	ping(registry string) error
+}
+
+// defaultRegistryPinger implements registryPinger.
+type defaultRegistryPinger struct {
+}
+
+func (drp *defaultRegistryPinger) ping(registry string) error {
+	healthzCheck := func(proto, registry string) error {
+		healthzResponse, err := http.Get(fmt.Sprintf("%s://%s/healthz", proto, registry))
```

Comment:
Hrm - I'm confused as to what we are doing here.  What's the rationale
behind a probe, vs failing the prune if the layers can't be removed?

On Jun 30, 2015, at 8:48 PM, Andy Goldstein <notifications@github.com>
wrote:

In pkg/image/prune/imagepruner.go
<https://github.com/openshift/origin/pull/3515#discussion_r33639990>:

>  }
>
>  var _ ImagePruner = &imagePruner{}
>
> +// registryPinger performs a health check against a registry.
> +type registryPinger interface {
> +	// ping performs a health check against registry.
> +	ping(registry string) error
> +}
> +
> +// defaultRegistryPinger implements registryPinger.
> +type defaultRegistryPinger struct {
> +}
> +
> +func (drp *defaultRegistryPinger) ping(registry string) error {
> +	healthzCheck := func(proto, registry string) error {
> +		healthzResponse, err := http.Get(fmt.Sprintf("%s://%s/healthz", proto, registry))

Oh ok. Is it possible to easily invoke an arbitrary liveness probe? What
I've got is just using http.Client by hand. Is a probe better in this use
case?

—
Reply to this email directly or view it on GitHub
<https://github.com/openshift/origin/pull/3515/files#r33639990>.